File: project-euler/009/problem009.rb
#!/usr/bin/env ruby
class Integer
  def squared
    self * self
  end
end

def is_pythagorean(a, b, c)
  a.squared + b.squared == c.squared
end

(0..1000).each do |a|
  (a..1000).each do |b|
    c = 1000 - a - b
    if is_pythagorean(a, b, c) && a + b + c == 1000
      puts a * b * c
    end
  end
end

File: project-euler/009/problem009.py
#!/usr/bin/env python
for a in range(1, 1000):
    for b in range(a, 1000):
        c = 1000 - a - b
        if a * a + b * b == c * c and a + b + c == 1000:
            print(a * b * c)